src/core/border_router/routing_manager.cpp

```diff
@@ -3681,6 +3681,13 @@ Error RoutingManager::PdPrefixManager::Process(const Ip6::Nd::RouterAdvertMessag
     {
         DiscoveredPrefixTable::Entry entry;
 
+
+        if (aMessage->GetHeader().GetPreferredLifetime()==0)
+        {
+            entry.mPrefix.Clear();
+            continue;
+        }
```

Comment:
Having this check in the `for()` loop does not make sense.

The `aMessage` is not changing (it is `const`) and therefore `aMessage->GetHeader().GetPreferredLifetime()` is also fixed, so this check will either always be `true` or `false` independent of the `option` from `for()` loop.